pkg/executor/aggregate/agg_spill.go

```diff
@@ -56,3 +322,112 @@ func (a *AggSpillDiskAction) Action(t *memory.Tracker) {
 func (*AggSpillDiskAction) GetPriority() int64 {
 	return memory.DefSpillPriority
 }
+
+// ParallelAggSpillDiskAction implements memory.ActionOnExceed for parallel HashAgg.
+type ParallelAggSpillDiskAction struct {
+	memory.BaseOOMAction
+	e           *HashAggExec
+	spillHelper *parallelHashAggSpillHelper
+	spillTimes  atomic.Int32
+}
+
+// Action set HashAggExec spill mode.
+func (p *ParallelAggSpillDiskAction) Action(t *memory.Tracker) {
+	if !p.actionImpl(t) {
+		return
+	}
+
+	if t.CheckExceed() {
+		p.triggerFallBackAction(t)
+	}
+}
+
+func (p *ParallelAggSpillDiskAction) checkRestriction(t *memory.Tracker) bool {
+	return hasEnoughDataToSpill(p.e.memTracker, t) && p.spillTimes.Load() < maxSpillTimes
```

Comment:
Why limit max spill times for parallel agg?

Reply:
> Why limit max spill times for parallel agg?

According to question 2 of https://asktug.com/t/topic/1005626/5?u=xuhuaiyu-pingcap, to avoid the long time occupied by hash agg spill, we set a `maxSpillTimes`.

Reply:
I think at least should make `maxSpillTimes` configurable.

Reply:
> I think at least should make `maxSpillTimes` configurable.

`maxSpillTimes` is removed now.